QRHunter/app/src/main/java/com/example/qrhunter/utils/PlayerUtil.java

```diff
@@ -52,64 +59,112 @@ public static Map<String, Object> convertPlayerToHashmap(Player player) {
     /**
      * Calculate the total score given the list of qr codes
      *
-     * @param qrCodes The qr codes where their score will be calculated
+     * @param user The reference to the user to obtain scores from
      * @return The sum of the scores
      */
-    public double calculateTotalScore(ArrayList<QRCode> qrCodes) {
-        double currentTotalScore = 0;
-        for (int i = 0; i < qrCodes.size(); i++) {
-            QRCode qrcode = qrCodes.get(i);
-            currentTotalScore = currentTotalScore + qrcode.getScore();
-        }
-
-        return currentTotalScore;
+    public static double calculateTotalScore(String user) {
+        final double[] currentTotalScore = {0};
+        FirebaseFirestore db = FirebaseFirestore.getInstance();
+        db.collection("players")
+                .whereEqualTo("username", user)
+                .get()
+                .addOnCompleteListener(new OnCompleteListener<QuerySnapshot>() {
+                    @Override
+                    public void onComplete(@NonNull Task<QuerySnapshot> task) {
+                        if (task.isSuccessful()) {
+                            for (QueryDocumentSnapshot document : task.getResult()) {
+                                Log.d("TAG", document.getId() + " => " + document.getData());
+                                Double i = (Double) document.getDouble("totalScore");
+                                currentTotalScore[0] += i;
+                                Log.d("TAG", String.valueOf(currentTotalScore[0]));
+                            }
+                        } else {
+                            Log.d("TAG", "Error getting documents: ", task.getException());
+                        }
+                    }
+                });
+
+
+        return currentTotalScore[0];
```

Comment:
This function will always return 0 because the `onComplete` function u overriden is asynchronous, meaning that the `onComplete` function is only called after the function returns. Related stackoverflow post [here](https://stackoverflow.com/questions/48499310/how-to-return-a-documentsnapshot-as-a-result-of-a-method?noredirect=1&lq=1). 

In fact, you don't need to call Firebase at all since ProfileViewModel will already have the scanned qr codes. I think you should revert all these changes and use your old logic. In `ProfileViewModel.java`, in the `getScannedQRCodes()` function, just call this calculate highest score function

Reply:
Sounds good thank you